rotkehlchen/balances/historical.py

```diff
@@ -74,6 +75,30 @@ def get_balances(self, timestamp: Timestamp) -> dict[Asset, HistoricalBalance]:
 
         return result
 
+    def get_erc721_tokens_balances(
+            self,
+            address: 'ChecksumEvmAddress',
+            assets: tuple[Asset, ...],
+    ) -> dict[Asset, FVal]:
+        """Get current balances for the given erc721 assets of a specific address by processing historical events.
+
+        May raise:
+            - NotFoundError if no events exist for the assets/address
+            - DeserializationError if there is a problem deserializing an event from DB
+        """  # noqa: E501
+        events, _ = self._get_events_and_currency(
+            assets=assets,
+            address=address,
+        )
+        if len(events) == 0:
+            raise NotFoundError(f'No historical data found for {assets} for user address {address}')  # noqa: E501
+
+        current_balances: dict[Asset, FVal] = defaultdict(FVal)
+        for event in events:
+            self._update_balances(event=event, current_balances=current_balances)
+
+        return {asset: balance for asset, balance in current_balances.items() if balance > ZERO}
```

Comment:
Since this is always ERC721 tokens please do the resolution here, and adjust the types.

Then you don't need to resolve EVM token in that loop in `_detect_erc721_tokens`. And this way you won't need to worry what if resolution fails, as it will be explicit it can't fail.